src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/LocationFiltersForm.test.tsx

```diff
@@ -0,0 +1,298 @@
+        },
+      ],
+    },
+  };
+
+  test('renders location group options correctly', () => {
+    render(
+      <LocationFiltersForm
+        {...testWizardStepProps}
+        options={testLocations}
+        onSubmit={noop}
+      />,
+    );
+
+    expect(
+      within(
+        screen.getByRole('group', {
+          name: 'Country',
+          hidden: true,
+        }),
+      ).getAllByRole('checkbox', {
+        hidden: true,
+      }),
+    ).toHaveLength(1);
+
+    expect(
+      within(
+        screen.getByRole('group', {
+          name: 'Local authority',
+          hidden: true,
+        }),
+      ).getAllByRole('checkbox', {
+        hidden: true,
+      }),
+    ).toHaveLength(3);
+
+    expect(
+      within(
+        screen.getByRole('group', {
+          name: 'Region',
+          hidden: true,
+        }),
+      ).getAllByRole('checkbox', {
+        hidden: true,
+      }),
+    ).toHaveLength(2);
+  });
+
+  test('selecting options shows the number of selected options for each location group', () => {
+    render(
+      <LocationFiltersForm
+        {...testWizardStepProps}
+        options={testLocations}
+        onSubmit={noop}
+      />,
+    );
+
+    userEvent.click(screen.getByLabelText('Local authority 2'));
+
+    expect(
+      screen.getByRole('button', {
+        name: 'Local authority - 1 selected',
+      }),
+    ).toBeInTheDocument();
+
+    userEvent.click(screen.getByLabelText('Local authority 3'));
+    userEvent.click(screen.getByLabelText('Country 1'));
+    userEvent.click(screen.getByLabelText('Region 2'));
+
+    expect(
+      screen.getByRole('button', {
+        name: 'Local authority - 2 selected',
+      }),
+    ).toBeInTheDocument();
+
+    expect(
+      screen.getByRole('button', {
+        name: 'Country - 1 selected',
+      }),
+    ).toBeInTheDocument();
+
+    expect(
+      screen.getByRole('button', {
+        name: 'Region - 1 selected',
+      }),
+    ).toBeInTheDocument();
+  });
+
+  test('shows validation errors if no location is selected', async () => {
+    render(
+      <LocationFiltersForm
+        {...testWizardStepProps}
+        options={testLocations}
+        onSubmit={noop}
+      />,
+    );
+
+    userEvent.click(
+      screen.getByRole('button', {
+        name: 'Next step',
+      }),
+    );
+
+    await waitFor(() => {
+      expect(
+        screen.getByRole('link', {
+          name: 'Select at least one location',
+        }),
+      ).toBeInTheDocument();
+    });
+  });
+
+  test('only the correct checkboxes are selected from initial values', () => {
+    render(
+      <LocationFiltersForm
+        {...testWizardStepProps}
+        options={testLocations}
+        onSubmit={noop}
+        initialValues={{
+          LocalAuthority: ['local-authority-1', 'local-authority-3'],
+          Region: ['region-2'],
+        }}
+      />,
+    );
+
+    expect(screen.getByLabelText('Local authority 1')).toHaveAttribute(
+      'checked',
```

Comment:
For all of these we could use the `toBeChecked` method - same for any other place in this test suite too.